modal/stub.py

```diff
@@ -870,32 +892,41 @@ def cls(
         cloud: Optional[str] = None,  # Cloud provider to run the function on. Possible values are aws, gcp, auto.
     ) -> type:
         function_handles: Dict[str, _FunctionHandle] = {}
+        partial_functions: Dict[str, _PartialFunction] = {}
+
+        if hasattr(user_cls, "remote"):
+            raise InvalidError(f"Class {user_cls} has an attribute named 'remote' [TODO: expand]")
+
+        kwargs = dict(
+            _cls=user_cls,
+            image=image,
+            secret=secret,
+            secrets=secrets,
+            gpu=gpu,
+            serialized=serialized,
+            mounts=mounts,
+            shared_volumes=shared_volumes,
+            allow_cross_region_volumes=allow_cross_region_volumes,
+            cpu=cpu,
+            memory=memory,
+            proxy=proxy,
+            retries=retries,
+            concurrency_limit=concurrency_limit,
+            container_idle_timeout=container_idle_timeout,
+            timeout=timeout,
+            interactive=interactive,
+            keep_warm=keep_warm,
+            cloud=cloud,
+        )
+
         for k, v in user_cls.__dict__.items():
             if isinstance(v, (PartialFunction, AioPartialFunction)):
-                partial_function = synchronizer._translate_in(v)  # TODO: remove need for?
-                function_handles[k] = self.function(
-                    _cls=user_cls,
-                    image=image,
-                    secret=secret,
-                    secrets=secrets,
-                    gpu=gpu,
-                    serialized=serialized,
-                    mounts=mounts,
-                    shared_volumes=shared_volumes,
-                    allow_cross_region_volumes=allow_cross_region_volumes,
-                    cpu=cpu,
-                    memory=memory,
-                    proxy=proxy,
-                    retries=retries,
-                    concurrency_limit=concurrency_limit,
-                    container_idle_timeout=container_idle_timeout,
-                    timeout=timeout,
-                    interactive=interactive,
-                    keep_warm=keep_warm,
-                    cloud=cloud,
-                )(partial_function)
+                partial_functions[k] = synchronizer._translate_in(v)  # TODO: remove need for?
+                function_handles[k] = self.function(**kwargs)(partial_functions[k])
 
         _PartialFunction.initialize_cls(user_cls, function_handles)
+
+        user_cls.remote = self._make_remote_cls(user_cls, partial_functions, kwargs)
```

Comment:
Right now `user_cls.remote` is a class constructor. I think it would make more sense that it's just a factory function that returns a `Remote` instance instead. The benefit of doing that is it can be async and do I/O. So you can have `user_cls` do the RPC call that resolves the bound arguments to some sort of "class id". It seems nice to do that just once per class instantiation.